### ISSUE TITLE
 Can't install qgis-plugin-grass : Depends: grass786

### ISSUE THREAD

**Issue by tazrart** [qgis/QGIS#49030]
### Feature description

Dear QGIS Users,

I wanted install QGIS (3.24) on Ubuntu (20.04) following the steps cited in : https://www.qgis.org/fr/site/forusers/alldownloads.html 
(section Debian/ubuntu), but I have this error  : qgis-plugin-grass : Depends: grass786.

 I tried to follow the solutions proposed in the issue below but I can't solve my problem:
https://github.com/qgis/QGIS/issues/38329

Here are my command lines in the terminal:

1. Remove QGIS and dependent packages :

```
sudo apt-get remove --auto-remove qgis-providers
sudo apt purge gdal* --auto-remove
sudo apt purge qgis* python-qgis* --auto-remove
sudo apt purge grass* --auto-remove
sudo apt purge proj* --auto-remove
sudo apt purge geos* --auto-remove
```

2. Install QGIS :

```
sudo apt install gnupg software-properties-common

wget -qO - https://qgis.org/downloads/qgis-2021.gpg.key | sudo gpg --no-default-keyring --keyring gnupg-ring:/etc/apt/trusted.gpg.d/qgis-archive.gpg --import

sudo chmod a+r /etc/apt/trusted.gpg.d/qgis-archive.gpg

sudo add-apt-repository "deb https://qgis.org/ubuntu $(lsb_release -c -s) main"

sudo apt update

sudo apt install qgis qgis-plugin-grass
```

3. But I have this error : 

```
Reading package lists... Done
Building dependency tree       
Reading state information... Done
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies:
 qgis-plugin-grass : Depends: grass786
E: Unable to correct problems, you have held broken packages.
```

How to solve this problem please?

Thank you in advance for your help.

Best regards,

**Comment by paulfouquet** [comment]
I have the same issue

**Comment by Uatschitchun** [comment]
Yes, me too ...

**ubuntugis** version depends on `grass820` and **ubuntu** version depends on `grass782`

both dependencies are not able to be resolved.
`grass` from _ubuntugis_ offers `grass78`

**Comment by Uatschitchun** [comment]
@agiudiceandrea
This refers, like the initial issue, to focal (20.04).
I know that focal does not support latest qgis version cause of QT 5.13, but being able to install latest QT 5.12 version (3.26?) would still be nice

**Comment by agiudiceandrea** [comment]
@Uatschitchun, I'm sorry but QGIS 3.26 and Ubuntu Focal are no longer supported:
https://github.com/qgis/QGIS-Website/commit/55e2b2e26164a63bb5ce9c5c35041cfa923c6997
https://qgis.org/en/site/getinvolved/development/roadmap.html

**Comment by aloboa** [comment]
Following https://www.qgis.org/resources/installation-guide/
I have this issue with ubuntu 22.04
$ sudo apt install qgis-plugin-grass
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies:
 qgis-plugin-grass : Depends: grass787
E: Unable to correct problems, you have held broken packages.

Please, reopen.

**Comment by agiudiceandrea** [comment]
> $ sudo apt install qgis-plugin-grass

@aloboa, shouldn't it be `sudo apt install qgis qgis-plugin-grass`?

**Comment by aloboa** [comment]
@agiudiceandrea `sudo apt install qgis` works fine. The problem is with qgis-plugin-grass only.

**Comment by agiudiceandrea** [comment]
@jef-n, have you seen this issue report?

**Comment by jef-n** [comment]
> @jef-n, have you seen this issue report?

Maybe, but it doesn't have enough information to be conclusive anyway.

```
$ cat pkgtest.sh
#!/bin/bash

set -e

dist=jammy

debootstrap --components=main,universe $dist $dist

mkdir -m755 -p $dist/etc/apt/keyrings
wget -O $dist/etc/apt/keyrings/qgis-archive-keyring.gpg https://download.qgis.org/downloads/qgis-archive-keyring.gpg

cat <<EOF >>$dist/etc/apt/sources.list
deb http://archive.ubuntu.com/ubuntu $dist-security main universe
deb http://archive.ubuntu.com/ubuntu $dist-updates main universe
EOF

cat <<EOF >$dist/etc/apt/sources.list.d/qgis.sources
Types: deb deb-src
URIs: https://qgis.org/debian
Suites: $dist
Architectures: amd64
Components: main
Signed-By: /etc/apt/keyrings/qgis-archive-keyring.gpg
EOF

chroot $dist apt-get update
chroot $dist apt-get install -y qgis-plugin-grass
chroot $dist dpkg -l "*grass*"
$ sudo pkgtest.sh
…
Processing triggers for libc-bin (2.35-0ubuntu3) ...
Desired=Unknown/Install/Remove/Purge/Hold
| Status=Not/Inst/Conf-files/Unpacked/halF-conf/Half-inst/trig-aWait/Trig-pend
|/ Err?=(none)/Reinst-required (Status,Err: uppercase=bad)
||/ Name                     Version          Architecture Description
+++-========================-================-============-=====================================================
un  grass                    <none>           <none>       (no description available)
ii  grass-core               7.8.7-1          amd64        GRASS GIS core components
un  grass-dev                <none>           <none>       (no description available)
un  grass-dev-doc            <none>           <none>       (no description available)
ii  grass-doc                7.8.7-1          all          GRASS GIS user documentation
un  grass-gui                <none>           <none>       (no description available)
un  grass787                 <none>           <none>       (no description available)
un  libgdal-grass            <none>           <none>       (no description available)
ii  libqgisgrass7-3.38.1     1:3.38.1+36jammy amd64        QGIS - shared grass library
ii  qgis-plugin-grass        1:3.38.1+36jammy amd64        GRASS plugin for QGIS
ii  qgis-plugin-grass-common 1:3.38.1+36jammy all          GRASS plugin for QGIS - architecture-independent data
ii  qgis-provider-grass      1:3.38.1+36jammy amd64        GRASS provider for QGIS
```

**Comment by jef-n** [comment]
BTW same for 1:3.34.9+36jammy

**Comment by agiudiceandrea** [comment]
I've just installed qgis 3.38.1 on Ubuntu Jammy Jellyfish 22.04 with `sudo apt install qgis qgis-plugin-grass` following https://qgis.org/resources/installation-guide/#debianubuntu without any issue.

**Comment by aloboa** [comment]
But not in my case, and this is well reported. Ignoring a problem is not fixing it,
please do not close it. The grass plugin is not really relevant for me, but it might be
for other people, and closing like this is not the way to achieve robust software.

```
(base) alobo@pop-os:~$ lsb_release -cs
jammy
(base) alobo@pop-os:~$ cat /etc/apt/sources.list.d/qgis.sources
Types: deb deb-src
URIs: https://qgis.org/debian
Suites: jammy
Architectures: amd64
Components: main
Signed-By: /etc/apt/keyrings/qgis-archive-keyring.gpg

(base) alobo@pop-os:~$ sudo apt update
Hit:1 https://cloud.r-project.org/bin/linux/ubuntu jammy-cran40/ InRelease
Hit:3 https://ppa.launchpadcontent.net/c2d4u.team/c2d4u4.0+/ubuntu jammy InRelease                     
Hit:4 https://ppa.launchpadcontent.net/yannubuntu/boot-repair/ubuntu jammy InRelease
Hit:5 http://apt.pop-os.org/proprietary jammy InRelease                
Hit:2 https://debian.qgis.org/debian jammy InRelease
Hit:6 http://apt.pop-os.org/release jammy InRelease
Hit:7 http://apt.pop-os.org/ubuntu jammy InRelease
Hit:8 http://apt.pop-os.org/ubuntu jammy-security InRelease
Hit:9 http://apt.pop-os.org/ubuntu jammy-updates InRelease
Hit:10 http://apt.pop-os.org/ubuntu jammy-backports InRelease
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
All packages are up to date.
(base) alobo@pop-os:~$ sudo apt install qgis qgis-plugin-grass
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
qgis is already the newest version (1:3.38.1+36jammy).
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies:
 qgis-plugin-grass : Depends: grass787
E: Unable to correct problems, you have held broken packages.
(base) alobo@pop-os:~$ 
```

**Comment by agiudiceandrea** [comment]
@aloboa, you are using Pop!_OS, not Ubuntu (as previously reported), aren't you?
It looks like your issue may have the same root cause of https://github.com/qgis/QGIS/issues/57951.

**Comment by dimzog** [comment]
Ubuntu 22.04 here. So what fixed it for me was using `Jammy` instead of `jammy` in `/etc/apt/sources.list.d/qgis.sources`

**Comment by edimilson-rodrigues** [comment]
I encountered a rather complicated path to resolve this issue. Initially, I installed QGIS on Ubuntu 22.04 following the official guide without any problems. However, I later needed to use GRASS within Python directly and made a few changes that inadvertently affected QGIS. When I attempted to clean up and reinstall QGIS from scratch, several issues arose.

The first issue was related to missing libraries (libpdal_base.so.13 and libpdal_util.so.13). I manually created symbolic links to existing versions by following the guidance from [this comment](https://github.com/qgis/QGIS/issues/52433#issuecomment-1792309483). While this allowed QGIS to open, many tools were still missing.

I tried multiple approaches to reinstall QGIS, but none were successful until I followed [this suggestion](https://github.com/qgis/QGIS/issues/49030#issuecomment-2480495096) to change the sources. I adjusted the source URLs from jammy to Jammy and added -lts to the URL. Even after this, another issue arose: Python could not access the QGIS modules (ModuleNotFoundError: No module named 'qgis').

To resolve this Python issue, I added the QGIS Python path to the Python environment using the following command:

bash
Copy code
export PYTHONPATH=/usr/lib/python3/dist-packages:$PYTHONPATH
To make this change permanent, I edited the ~/.bashrc file:

bash
Copy code
sudo nano ~/.bashrc
I added the following line at the end of the file:

bash
Copy code
export PYTHONPATH=/usr/lib/python3/dist-packages:$PYTHONPATH
Then, I reloaded the configuration:

bash
Copy code
source ~/.bashrc
Finally, QGIS worked perfectly. However, I had to revert the source URLs back to jammy to ensure normal updates via the package manager.
